File: Challenges/450_DSA_Cracker/Array/001_Reverse_The_Array.cpp
#include <bits/stdc++.h> 
typedef long long int lli;

void reverseArray(vector<int> &arr , int m)
{
	int beg = m + 1, end = arr.size() - 1;
	int mid = (beg + end) / 2; 
	lli a, b;
	
	for(int i = beg; i < mid + 1; i++) {
		a = arr[i];
		b = arr[end - (i-beg)];
		arr[i] = b;
		arr[end - (i-beg)] = a;
	}

	return;
}

File: Challenges/450_DSA_Cracker/Array/001_Reverse_The_Array.java
import java.util.* ;
import java.io.*; 
import java.util.ArrayList;

public class Solution 
{
    public static void reverseArray(ArrayList<Integer> arr, int m)
    {
        int beg = m + 1;
        int end = arr.size() - 1;
        int mid = (beg + end) / 2;
        int a, b;

        for(int i = beg; i < mid + 1; i++) {
            a = arr.get(i);
            b = arr.get(end - (i - beg));
            arr.set(i, b);
            arr.set(end - (i - beg), a);
        } 
    }
}
